paddle/fluid/platform/mkldnn_helper.h

```diff
@@ -251,5 +251,17 @@ inline mkldnn::memory::format MKLDNNFormatForSize(
   return data_format;
 }
 
+inline mkldnn::memory::format data_format_to_memory_format(
+    const std::string& data_format) {
+  switch (framework::StringToDataLayout(data_format)) {
```

Comment:
Did you check, what is happen when the input format is exactly the same as `framework::DataLayout::kNHWC`?. Does the performance will be the same before and after those changes?

Reply:
Why does it matter? I think it is AnyLayout always except in tests where I request NCHW.
I tested ResNet script and haven't seen change in fps.